packages/dataviews/src/style.scss

```diff
@@ -421,27 +421,47 @@
 		}
 
 		.dataviews-view-list__item-actions {
+			position: absolute;
+			top: $grid-unit-20;
+			right: 0;
+
+
+			> div {
+				height: $button-size-small;
+			}
+
 			.components-button {
 				opacity: 0;
-				position: fixed;
-				right: 0;
+			}
+		}
+
+		&:has(.dataviews-view-list__fields:empty) {
+			.dataviews-view-list__item-actions {
+				top: 50%;
+				transform: translateY(-50%);
 			}
 		}
 
 		&.is-selected,
 		&.is-hovered,
 		&:focus-within {
 			.dataviews-view-list__item-actions {
-				padding-right: $grid-unit-30;
+				background: #f8f8f8;
+				padding-left: $grid-unit-10;
+				margin-right: $grid-unit-30;
+				box-shadow: -12px 0 8px 0 #f8f8f8;
```

Comment:
Can we substitute a few variables here?

Reply:
We have an outstanding todo item for this gray. It will mean expanding the existing scale a bit, so needs a dedicated PR. 

I'm not sure variables really necessary for shadows.